### DISCUSSION TITLE
Is possible this function?

### DISCUSSION THREAD

**Discussion by Pablete01** [supabase/supabase#22542] [Questions] [answered]
Hi. I Have a file in storage, called "Precios.json". I like to read the file and put the data into a table in supabase. I try with this function:

DECLARE
    json_data JSONB;
BEGIN
    SELECT json_data FROM storage.objects WHERE bucket_id = 'data' AND name = 'precios.json' INTO json_data;

    INSERT INTO oaprecios (idempresa, producto, precio, fecha)
    SELECT (data->>'idempresa')::INT,
           (data->>'producto'),
           (data->>'precio')::NUMERIC(10, 2) AS precio,
           (data->>'fecha')
    FROM jsonb_array_elements(json_data) AS data;

END;

but json_data appear empty. Is possible to do something like this?
Thanks

**Comment by GaryAustin1** [comment, marked as answer]
storage.objects does not contain file data, just file metadata.  It is a postgres table.
You have to use the storage api calls to download data, or use a storage url to fetch the data.

**Reply by Pablete01** [reply]
Hi. I understand. Thanks, i'm newbie in this.
